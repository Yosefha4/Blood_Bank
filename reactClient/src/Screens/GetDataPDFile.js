import React from "react";
import PDFfile from "../components/PDFfile";
import { PDFDownloadLink } from "@react-pdf/renderer";

const GetDataPDFile = () => {
  const currentUserType = window.localStorage.getItem("userType");
  console.log(currentUserType);

  return (
    <div
      style={{
        alignItems: "center",
        textAlign: "center",
        marginTop: 36,
        justifyContent: "center",
      }}
    >
      {currentUserType === "Admin" ? (
        <div>
         
          <h3 style={{ marginBottom: 24 }}>PDF EXPORT<br />
          </h3>
          <h4 style={{ marginBottom: 12 }}>Get The History Action :</h4>{" "}
          <PDFDownloadLink document={<PDFfile />} fileName="Action History">
            {({ loading }) =>
              loading ? (
                <button
                  style={{
                    backgroundColor: "lightblue",
                    borderRadius: 6,
                    width: 120,
                    height: 40,
                    fontWeight: "bold",
                  }}
                >
                  Loading ...
                </button>
              ) : (
                <button
                  style={{
                    backgroundColor: "lightblue",
                    borderRadius: 6,
                    width: 120,
                    height: 40,
                    fontWeight: "bold",
                  }}
                >
                  Download PDF
                </button>
              )
            }
          </PDFDownloadLink>{" "}
        </div>
      ) : (
        <div style={{ fontWeight: "bold", fontSize: 24 }}>
          Requires Admin Permission ! <br /> Only an 'Admin' type user is
          authorized to access this information...
        </div>
      )}
      {/* <PDFfile /> */}
    </div>
  );
};

export default GetDataPDFile;
