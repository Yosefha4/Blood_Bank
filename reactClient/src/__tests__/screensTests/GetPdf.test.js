import React from "react";
import GetDataPDFile from "../../Screens/GetDataPDFile";
import { render, screen, waitFor } from "@testing-library/react";
import { PDFDownloadLink } from "@react-pdf/renderer"; // Import the PDFDownloadLink component

// import "jest-localstorage-mock";

// Mock the react-pdf library
jest.mock("@react-pdf/renderer", () => ({
  StyleSheet: {
    create: (styles) => styles,
  },
  Document: "div",
  Page: "div",
  Text: "span",
}));

// // Mock the localStorage
// const mockLocalStorage = (() => {
//   let store = {};

//   return {
//     getItem: jest.fn((key) => store[key] || null),
//     setItem: jest.fn((key, value) => {
//       store[key] = value.toString();
//     }),
//     removeItem: jest.fn((key) => {
//       delete store[key];
//     }),
//     clear: jest.fn(() => {
//       store = {};
//     }),
//   };
// })();

// Object.defineProperty(window, "localStorage", {
//   value: mockLocalStorage,
//   writable: true,
// });

// Mock the PDFDownloadLink component
// jest.mock("@react-pdf/renderer", () => ({
//   PDFDownloadLink: ({ document, fileName }) => (
//     <a href="#" download={`${fileName}.pdf`}>
//       Download PDF
//     </a>
//   ),
// }));


// jest.mock("../../components/PDFfile", () => {
//   return () => <div>Mocked PDFfile component</div>;
// });



describe("Should render GetPdfData correctlly", () => {
  test("Render 'Permmission Warning'  on Student/Employee UserType", () => {
    // const userTypeAdmin = "Admin";
    window.localStorage.setItem("userType", "Employee");

    render(<GetDataPDFile />);

    const adminPermissionMessage = screen.getByText(
      /Requires Admin Permission/i
    );
    expect(adminPermissionMessage).toBeDefined();
  });

});
// describe("Should render GetPdfData on Admin correctlly", () => {

 
//   test("Render 'the titles correct", () => {

//     const mockLocalStorage = {
//       getItem: jest.fn(() =>  'Admin'),
//       setItem: jest.fn(),
//       removeItem: jest.fn(),
//     };
//     Object.defineProperty(window, "localStorage", {
//       value: mockLocalStorage,
//       writable: true,
//     });

 

//     mockLocalStorage.setItem("userType", "Admin");
//     console.log("first", mockLocalStorage.getItem("userType"));

//     render(<GetDataPDFile />);

//     const adminPermissionMessage = screen.getByText(
//       /PDF EXPORT/i
//     );
//     expect(adminPermissionMessage).toBeDefined();

//   });

// });
