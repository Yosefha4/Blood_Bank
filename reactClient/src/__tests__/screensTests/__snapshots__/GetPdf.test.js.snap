// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render GetPdfData correctlly Render 'Download PDF' button on Admin UserType 1`] = `[Function]`;
